syft/frameworks/torch/he/fv/evaluator.py

```diff
@@ -0,0 +1,88 @@
+import copy
+
+from syft.frameworks.torch.he.fv.util.operations import poly_add_mod
+from syft.frameworks.torch.he.fv.util.operations import multiply_add_plain_with_delta
+from syft.frameworks.torch.he.fv.ciphertext import CipherText
+from syft.frameworks.torch.he.fv.plaintext import PlainText
+from syft.frameworks.torch.he.fv.integer_encoder import IntegerEncoder
+
+
+class Evaluator:
+    def __init__(self, context):
+        self.context = context
+        self.coeff_modulus = context.param.coeff_modulus
+
+    def add(self, op1, op2):
+        """Adds two operands using FV scheme.
+
+        Args:
+            op1 (Ciphertext/Plaintext): First argument.
+            op2 (Ciphertext/Plaintext): Second argument.
+
+        Returns:
+            If both arguments are Plaintext elements then the result will be a Plaintext object
+                otherwise a Ciphertext object with value equivalent to the result of addition
+                operation of two provided arguments.
+        """
+        if isinstance(op1, CipherText) and isinstance(op2, CipherText):
+            return self._add_cipher_cipher(op1, op2)
+
+        elif isinstance(op1, PlainText) and isinstance(op2, PlainText):
+            return self._add_plain_plain(op1, op2)
+
+        elif isinstance(op1, PlainText) and isinstance(op2, CipherText):
+            return self._add_plain_cipher(op1, op2)
+
+        elif isinstance(op1, CipherText) and isinstance(op2, PlainText):
+            return self._add_plain_cipher(op2, op1)
+
+        else:
+            raise TypeError(f"Addition Operation not supported between {type(op1)} and {type(op2)}")
+
+    def _add_cipher_cipher(self, ct1, ct2):
+        """Adds two ciphertexts.
+
+        Args:
+            ct1 (Ciphertext): First argument.
+            ct2 (Ciphertext): Second argument.
+
+        Returns:
+            A Ciphertext object with value equivalent to result of addition of two provided
+                arguments.
+        """
+        ct1, ct2 = copy.deepcopy(ct1.data), copy.deepcopy(ct2.data)
+        result = ct2 if len(ct2) > len(ct1) else ct1
+
+        for i in range(min(len(ct1), len(ct2))):
```

Comment:
Q: does this mean that if we have two ciphertext `c1=(c1[0], c1[1])` and `c2=(c2[0], c2[1], c2[2])`, the result would be `c=(c1[0] + c2[0], c1[1] + c2[1])`?

Reply:
extra c2[2] will be already present in the result by the:
`result = ct2 if len(ct2) > len(ct1) else ct1`